### DISCUSSION TITLE
PostgREST foreign key queries

### DISCUSSION THREAD

**Discussion by imknight** [supabase/supabase#3370] [Questions] [answered]
Hi looking for some help on how to query the API

1. how do I query with API for SQL  "Select * from projects where user_id in ( select * from users )"

2. how do I query with the relation key? I manage to have  <supabasedomain>/rest/v1/projects?select=*,development(name) to get all projects with development, how can I match it with development name?

**Comment by dshukertjr** [comment, marked as answer]
Hi @imknight!

You can query 1 by creating a [view](https://supabase.io/blog/2020/11/18/postgresql-views). Once you create a view, you can query it from Supabase as if it were just a regular table. 

For number 2, could you clarify your question? In SQL, what would your query look like?

Just curious, but is there a reason why you are using the rest API and not the [client](https://supabase.io/docs/reference/javascript/supabase-client)?

**Reply by imknight** [reply]
thanks the answer for question 1. building integration for supabase and Adalo, that's why need to direct deal with API

**Reply by imknight** [reply]
```
CREATE TABLE "public"."development" (
    "id" int8 NOT NULL,
    "name" text NOT NULL,
    PRIMARY KEY ("id")
);
```
```
CREATE TABLE "public"."projects" (
    "id" int8 NOT NULL,
    "lot" text,
    "project_name" varchar,
    "created_at" timestamp DEFAULT now(),
    "development_id" int8,
    CONSTRAINT "projects_development_id_fkey" FOREIGN KEY ("development_id") REFERENCES "public"."development"("id"),
    PRIMARY KEY ("id")
);
```
for some reason I can get only get the development name from my other end, how can I query all the projects that belong to a specific development name? because it already using the project API, I need to use the project API to filter the development name.

**Reply by thebengeu** [reply]
You should be able to do something like this: `/rest/v1/development?select=name,projects(*)&name=eq.abc`.

**Reply by imknight** [reply]
the reason is the project API has been using as the main list ( pulling by nocode tools ), the options are limited that I can only add a filter on top for the project API.

**Comment by thorwebdev** [comment]
@imknight can you provide some more background on 1.? Are there projects that don't have a user_id? Is there an easier way to express the logic you're looking for?

**Reply by imknight** [reply]
nope because is one to many, the project can have multiple users
